packages/richtext-lexical/src/packages/@lexical/markdown/MarkdownImport.ts

```diff
@@ -255,28 +264,26 @@ function createTextFormatTransformersIndex(
     const tagRegExp = tag.replace(/([*^+])/g, '\\$1')
     openTagsRegExp.push(tagRegExp)
 
-    if (IS_SAFARI || IS_IOS || IS_APPLE_WEBKIT) {
+    // Single-char tag (e.g. "*"),
+    if (tag.length === 1) {
       fullMatchRegExpByTag[tag] = new RegExp(
-        `(${tagRegExp})(?![${tagRegExp}\\s])(.*?[^${tagRegExp}\\s])${tagRegExp}(?!${tagRegExp})`,
+        `(?<![\\\\${tagRegExp}])(${tagRegExp})((\\\\${tagRegExp})?.*?[^${tagRegExp}\\s](\\\\${tagRegExp})?)((?<!\\\\)|(?<=\\\\\\\\))(${tagRegExp})(?![\\\\${tagRegExp}])`,
       )
     } else {
+      // Multi‐char tags (e.g. "**")
       fullMatchRegExpByTag[tag] = new RegExp(
-        `(?<![\\\\${tagRegExp}])(${tagRegExp})((\\\\${tagRegExp})?.*?[^${tagRegExp}\\s](\\\\${tagRegExp})?)((?<!\\\\)|(?<=\\\\\\\\))(${tagRegExp})(?![\\\\${tagRegExp}])`,
+        `(?<!\\\\)(${tagRegExp})((\\\\${tagRegExp})?.*?[^\\s](\\\\${tagRegExp})?)((?<!\\\\)|(?<=\\\\\\\\))(${tagRegExp})(?!\\\\)`,
       )
     }
   }
 
   return {
     // Reg exp to find open tag + content + close tag
     fullMatchRegExpByTag,
-    // Reg exp to find opening tags
-    openTagsRegExp: new RegExp(
-      (IS_SAFARI || IS_IOS || IS_APPLE_WEBKIT ? '' : `${escapeRegExp}`) +
-        '(' +
-        openTagsRegExp.join('|') +
-        ')',
-      'g',
-    ),
+
+    // Regexp to locate *any* potential opening tag (longest first).
+    // eslint-disable-next-line regexp/no-useless-character-class, regexp/no-empty-capturing-group, regexp/no-empty-group
+    openTagsRegExp: new RegExp(`${escapeRegExp}(${openTagsRegExp.join('|')})`, 'g'),
     transformersByTag,
```

Comment:
Could you explain the changes in this file? Why isn't `IS_SAFARI || IS_IOS || IS_APPLE_WEBKIT` used anymore, which is all the consecutive `\\\\\\`, etc.?

Reply:
`IS_SAFARI || IS_IOS || IS_APPLE_WEBKIT ` was used because negative regex lookbehind was not well-supported by Safari back when this code was written (3 years ago). Now it is: https://caniuse.com/?search=Negative%20Lookbehind

\\\\\\\\\\\\\\\\ would turn into \\\\. JavaScript needs \\\\ if you want a literal backslash, and regex then also needs \\\\ on top of that, if you want to match a backslash.